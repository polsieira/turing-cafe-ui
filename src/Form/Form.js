import React, { Component } from 'react';
import '../Form/Form.scss';

class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      date: '',
      time: '',
      number: ''
    }
  }

  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value })
  }

  handleClick(event) {
    event.preventDefault();
    this.props.addReservation(this.state);
  }

  render() {
    return (
      <>
        <input type='text' name='name' value={this.state.name} placeholder='Name' onChange={(event) => { this.handleChange(event) }} />
        <input type='text' name='date' value={this.state.date} placeholder='Date (mm/dd)' onChange={(event) => { this.handleChange(event) }} />
        <input type='text' name='time' value={this.state.time} placeholder='Time' onChange={(event) => { this.handleChange(event) }} />
        <input type='text' name='number' value={this.state.number} placeholder='Number of guests' onChange={(event) => { this.handleChange(event) }} />
        <button type='submit' onClick={(event) => { this.handleClick(event) }}>Make Reservation</button>
      </>
    )
  }
}

export default Form;
