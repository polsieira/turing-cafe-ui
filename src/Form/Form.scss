input,
button {
  height: 30px;
  width: 15%;
}

input {
  font-size: 20px;
  padding: 5px;
}

button {
  background: teal;
  color: white;
  font-size: 12px;
  &:hover {
    cursor: pointer;
  }
}