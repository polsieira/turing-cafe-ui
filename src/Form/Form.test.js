import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Form from './Form';


describe('Form', () => {

  let wrapper;
  const mockAddReservation = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Form addReservation={mockAddReservation} />)
  })

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should update call handleChange when input is changed ** test simulates event **', () => {
    //Setup
    wrapper.instance().handleChange = jest.fn();
    const mockEvent = {
      target: {
        name: 'name',
        value: 'Pol'
      }
    }

    //Execution
    wrapper.find('input').at(0).simulate('change', mockEvent);

    //Expectation
    expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockEvent)
  })

  it('should update state when handleChange is called ** test changes state **', () => {
    //Setup
    const mockEvent = {
      target: {
        name: 'name',
        value: 'Pol'
      }
    }

    const newStateName = 'Pol';

    //Execution
    wrapper.instance().handleChange(mockEvent);

    //Expectation
    expect(wrapper.state('name')).toEqual(newStateName);
  })

  it('should update call addReservation when button is clicked ** test simulates event **', () => {
    //Setup
    wrapper.instance().handleClick = jest.fn();
    const mockEvent = {
      preventDefault: jest.fn()
    }

    //Execution
    wrapper.find('button').simulate('click', mockEvent);

    //Expectation
    expect(wrapper.instance().handleClick).toHaveBeenCalledWith(mockEvent)
  })

  it('should call addReservation when handleClick is called ** test changes state **', () => {
    //Setup
    wrapper.instance().clearInputs = jest.fn();
    const mockEvent = {
      preventDefault: jest.fn()
    }
    const newState = {
      name: 'Pol',
      date: '10/17',
      time: '7:00',
      number: '5'
    };
    wrapper.setState(newState)

    //Execution 
    wrapper.instance().handleClick(mockEvent)

    //Expectation
    expect(mockAddReservation).toHaveBeenCalledWith(newState);
    expect(wrapper.instance().clearInputs).toHaveBeenCalled();
  });

  it('should clear state when clearInputs is called ** test changes state', () => {
    //Setup
    const state = {
      name: 'Pol',
      date: '10/17',
      time: '7:00',
      number: '5'
    }
    wrapper.setState(state);
    const clearedState = {
      name: '',
      date: '',
      time: '',
      number: ''
    };

    //Execution 
    wrapper.instance().clearInputs();

    //Expectation
    expect(wrapper.state()).toEqual(clearedState)

  })
});