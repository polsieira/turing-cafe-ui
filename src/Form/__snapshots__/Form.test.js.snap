// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match a snapshot 1`] = `
<Fragment>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date (mm/dd)"
    type="text"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
    value=""
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of guests"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
    type="submit"
  >
    Make Reservation
  </button>
</Fragment>
`;
